### ISSUE TITLE
Invalid characters in filenames

### ISSUE THREAD

**Issue by grochamou** [justin025/onthespot#20]
### Bug Description

When downloading tracks with “/” in the title, the generated file name contains the “/” character too, and sub-folders are created.

The “/” character should never be part of a file name (but Python seems to allow it). If the track name contains a path separator (such as “:”, “/” or “\”), it must be deleted or replaced by a valid character (such as a space or underscore).

For example, https://open.spotify.com/track/4KkTFKkHEm523C3xptOtZx is resulting in the following structure:

"8. Christmas Hymns Medley O, Little Town of Bethlehem "
--- "Silent Night "
------ "Hark! the Herald Angels Sing "
--------- "Away in a Manger "
------------ "Joyful, Joyful, We Adore Thee "
--------------- " Joy to the World.mp3"

### Environment Details

```markdown
| **Detail**               | **Your Information**                 |
|--------------------------|--------------------------------------|
| **Operating System**     | MacOS 15.0.1                         |
| **Application Version**  | v0.7.1                               |
| **Installation Method**  | Build from Script                    |
| **Python Version**       | 3.12.7                               |
| **FFmpeg Version**       | 7.1                                  |
```

**Comment by justin025** [comment]
Should be fixed with the latest commit, if it does not fix your issue please let me know.

**Comment by dimitar2** [comment]
@justin025 Hi, I was going to report a new bug that the "/" character is missing from the MP3 tag title, but I thought this might be related. It happens with the latest 0.7.1. Can it be preserved in the tags while still being removed from the filename?

**Comment by justin025** [comment]
@dimitar2 try the latest dev build it replaces illegal characters with whatever you specify in advanced settings

https://github.com/justin025/onthespot/releases/tag/dev

**Comment by dimitar2** [comment]
@justin025 Thanks but my idea was to actually keep the "/" where it is allowed in the tag itself, as opposed to the filename where it definitely must be omitted or replaced. There are some song titles and band names like AC/DC where it would be good to have all true characters in the tags. Also thank you for supporting this project, it does an amazing job!

**Comment by justin025** [comment]
Perhaps there is a misunderstanding, the files metadata tags contain the original values from spotify ie AC/DC but the file iteslf will be named AC-DC.

This feature is already included in the latest dev build please check it out and let me know if there are any issues

**Comment by dimitar2** [comment]
@justin025 Yes, I tried it now and it works well for the tags. The illegal character replacement in filenames worked for me initially, but then I started seeing odd behavior.
For example, setting the Track Path as "{artist} - {name}", and illegal character replacement as "-", and downloading a song that contains a "/" in the title causes the download to fail.
Another example is by using the same settings but this time the track has a "/" in the artist name. In this case, the "/" is not replaced but is instead interpreted as a separator and an "AC" folder is created.
